keps/sig-apps/4368-support-managed-by-label-for-batch-jobs/README.md

```diff
@@ -0,0 +1,1193 @@
+-->
+
+Yes, we will have unit tests for the feature enablement / disablement after the
+Job is created (see [unit tests](#unit-tests)).
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+The rollout will not impact already running workloads, unless they set the
+"managed-by" label to a custom value, but this would require a prior intentional
+action.
+
+###### What specific metrics should inform a rollback?
+
+A substantial increase in the `apiserver_request_total[code=409, resource=job, group=batch]`,
+while there are jobs with the custom "managed-by" label, can be indicative of
+the built-in job controller stepping onto another controller, causing conflicts.
+This can be further investigate per-job by checking the `.metadata.managedFields.manager`
+being flipped between two owners.
+
+The feature is opt-in so in case of such problems the custom "managed-by" label
+should not be used.
+
+Also, an admin could check if the value of the `job_by_external_controller_total`
+matches the expectations. For example, if the value of the metric does not increase
+when new jobs are being added with a custom "managed-by" label, it might be
+indicative that the feature is not working correctly.
+
+A substantial increase in `kube_cronjob_status_active` after upgrade may suggest
+that the Jobs are not making progress. Additionally, if the non-progressing
+Jobs use custom "managed-by" label, then rollback of the feature might be
+justified to make the CronJobs run, by letting the built-in Job controller
+handle the Jobs.
+
+A substantial drop in the `job_sync_duration_seconds`, while the number of
+jobs with the custom "managed-by" label is low, could be indicative of the
+Job controller skipping reconciliation of jobs it should reconcile. This could
+be further investigated per-job by looking at the timestamp of changes in
+`.metadata.managedFields.time`, and owners in `.metadata.managedFields.manager`.
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+It will be tested manually prior to beta launch.
```

Comment:
Can you (at least briefly) describe the test scenario here?

Reply:
Sure, updated. PTAL.